hatchery/hatchery.go

```diff
@@ -496,7 +496,7 @@ func terminate(w http.ResponseWriter, r *http.Request) {
 		Config.Logger.Printf(userlicerr.Error())
 	}
 	Config.Logger.Printf("Debug: Active gen3 license user maps %v", activeGen3LicenseUsers)
-	if len(*activeGen3LicenseUsers) == 0 {
+	if activeGen3LicenseUsers == nil || len(*activeGen3LicenseUsers) == 0 {
```

Comment:
This would work, but I think the `getLicenseUserMapsForUser` function should return a pointer to an empty list instead of nil when the user doesn't have a license.
And the edge case should be added to the `Test_GetActiveGen3LicenseUserMaps` unit test